### ISSUE TITLE
Question about your code (forward_test() of attn_cycle_gan_model.py when opt.concat='rmult')

### ISSUE THREAD

**Issue by gymoon10** [voidstrike/ASGIT#5]
Hi, I'm Goo-Young Moon, a master's degree student at Korea University. I read your paper interestingly while investigating the research on synthetic data generation with GAN. But, I opened the issue as I had some questions while I was looking at the code you uploaded.

I trained the model with PHA & RHP(opt.concat=rmult). And I found that in the `forward_test` of `attn_cycle_model.py`, fake images are generated by following codes.
`self.fake_B = self.netG_A(self.real_A * 1.5)`
`self.fake_A = self.netG_B(self.real_B * 1.5)`

It seems strange that attention maps are not used unlike training case. Eventhough, the generators output plausiable translated images I wonder why you multiplied by 1.5 instead of attention maps which can be calculated by discriminators.

I'm sorry to bother you, but I'd appreciate your help. Thank you for your research and hard work.

**Comment by voidstrike** [comment]
Hey Goo-Young, thanks for your interest in our paper. I will try to answer your question and sorry ahead if I forgot something, the project was done 3 years ago.

Back to your question, if you look at the `forward` function, we trained the model with 
```
self.fake_B = self.netG_A(self.real_A * (1. + self.attn_A))
self.rec_A = self.netG_B(self.fake_B)
self.fake_A = self.netG_B(self.real_B * (1. + self.attn_B))
self.rec_B = self.netG_A(self.fake_A)
```
It may sound a little weird, yet we assume the whole image is important during the inference stage. To be compatible with the training scenario that includes an attention map, we manually inject a dummy attention map. That's where the 0.5 come from --> it plays the role of an attention map. It's worth noting that we don't generate attn_map during the inference.

To be honest, looking back at this paper 3 years later, it has some drawbacks: 1) The inconsistency of training & inference -- and you asked about it. A more natural & intuitive way of doing this would be FAL and CascadeRCNN (I know this is for object detection, but they share the same feedback mechanism); 2) The backbone is quite old and there are many SOTA GAN frameworks, achieving amazing performance. 

Anyway, all I want to say is that the core idea of this paper is to let the discriminator provide detailed feedback, but the way we did that is kind of out-of-date. :)
